tests/model/test_kfold.py

```diff
@@ -0,0 +1,31 @@
+from random import randrange
+
+import pandas as pd
+
+
+class Kfold:
+    def split(
```

Comment:
We need to make sure that this works when streaming data. Sources have an `async` `records()` method which yields one record at a time. We have to make sure any method we use doesn't require the whole dataset be in memory at the same time